src/Execution/Arguments/ArgPartitioner.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+namespace Nuwave\Lighthouse\Execution\Arguments;
+
+use ReflectionClass;
+use ReflectionNamedType;
+use Illuminate\Database\Eloquent\Model;
+use Nuwave\Lighthouse\Execution\ArgumentResolver;
+use Illuminate\Database\Eloquent\Relations\HasOne;
+use Nuwave\Lighthouse\Support\Contracts\Directive;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\MorphOne;
+use Illuminate\Database\Eloquent\Relations\MorphMany;
+use Illuminate\Database\Eloquent\Relations\MorphToMany;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+class ArgPartitioner
+{
+    /**
+     * Partition the arguments into nested and regular.
+     *
+     * @param  \Nuwave\Lighthouse\Execution\Arguments\ArgumentSet  $argumentSet
+     * @param  mixed  $root
+     * @return \Nuwave\Lighthouse\Execution\Arguments\ArgumentSet[]
+     */
+    public static function nestedArgumentResolvers(ArgumentSet $argumentSet, $root): array
+    {
+        $model = $root instanceof Model
```

Comment:
This is not `$model`, it should be `$modelReflection`

Reply:
Renamed.